ScoutSuite/providers/azure/provider.py

```diff
@@ -2,29 +2,30 @@
 
 import os
 
-import google.auth
 from opinel.utils.console import printError, printException
 
 from ScoutSuite.providers.base.provider import BaseProvider
 from ScoutSuite.providers.azure.configs.services import AzureServicesConfig
 
-from azure.common.credentials import ServicePrincipalCredentials
+from azure.common.client_factory import get_azure_cli_credentials
 
-class AzureCredentials():
+
+class AzureCredentials:
 
     def __init__(self, credentials, subscription_id):
         self.credentials = credentials
         self.subscription_id = subscription_id
 
+
 class AzureProvider(BaseProvider):
     """
-    Implements provider for AWS
+    Implements provider for Azure
     """
 
     def __init__(self, project_id=None, organization_id=None,
                  report_dir=None, timestamp=None, services=[], skipped_services=[], thread_config=4, **kwargs):
 
-        self.profile = 'aws-profile'  # TODO this is aws-specific
+        self.profile = 'azure-profile'  # TODO this is aws-specific
```

Comment:
Is that comment still relevant?

Reply:
Yes it is! self.profile and self.aws_account_id are members that are only really useful for AWS provider, but breaks if removed from other the other providers. Note that these are in GCP too.

self.profile is used in base/provider.py to generate report and has a comment saying that it would be a significant re-write to get rid of it, so I think that might be better to do a specific PR for this refactoring.

Reply:
Ok, no worries!